src/components/FormInput/FormInput.tsx

```diff
@@ -0,0 +1,63 @@
+import { useFormContext } from 'react-hook-form';
+import {
+  ErrorMessage,
+  Input,
+  InputContainer,
+  Label,
+  SuccessMessage,
+  StyledTitle
+} from './FormInput.style';
+
+interface FormInputProps {
+  name: string;
+  type: string;
+  success: boolean;
+  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  placeholder: string;
+  title: string;
+  errorMessage: string;
+  successMessage: string;
+  successColor: string;
+  errorColor: string;
+  show: boolean;
+}
+
+const FormInput = ({
+  name,
+  type = 'text',
+  title,
+  success,
+  placeholder,
+  errorMessage,
+  successMessage,
+  successColor,
+  errorColor,
+  show = false,
+  handleChange
+}: Partial<FormInputProps>) => {
+  const { register } = useFormContext();
+
+  return (
+    <InputContainer>
+      <Label>
+        <StyledTitle>{title}</StyledTitle>
+        {show && success && (
+          <SuccessMessage color={successColor}>{successMessage}</SuccessMessage>
+        )}
+        {show && !success && (
+          <ErrorMessage color={errorColor}>{errorMessage}</ErrorMessage>
+        )}
```

Comment:
show는 묶어서 관리할 수 있을 것 같아요!
```suggestion
        {show && (
  <>
    {success && <SuccessMessage color={successColor}>{successMessage}</SuccessMessage>}
    {!success && <ErrorMessage color={errorColor}>{errorMessage}</ErrorMessage>}
  </>
)}

```